Align naming of WAN replication classes and interfaces

mmedenjak opened the pull request:
Best-effort attempt to align naming of WAN classes and interfaces. The
convention is that WAN classes should try to start with "Wan". An
exception might be when we are talking about an abstract class, in which
case the name might start with "AbstractWan". To try to cover as much as
possible but to also constraint the amount of changes, I have imposed
some rules:
- no naming changes that might affect changing the names in XML and YAML
configuration. This would then cause changes in the parsers, many more
changes in tests, code samples, reference manual and so on.
- internal classes and tests do not necessarily need to be renamed right
now. We can rename then in any minor release or even patch release.
- no behaviour changes and no code changes except for renaming

Because of these rules, we leave the name "WanReplicationConfig" as well
as "WanReplicationService", amongst a few others. We also leave the
getters and setters to include the word "Wan", even though it is implied.
Fixing this this would cause changes in XML and YAML and I don't think
the benefit of not having the unnecessary word outweighs the change
needed to address it.

Changes in OS:
WanBatchReplicationPublisherConfigTest -> WanBatchPublisherConfigTest
CustomWanPublisherConfigTest -> WanCustomPublisherConfigTest
WanReplicationTest -> WanCustomPublisherTest
DummyWanReplication -> WanDummyPublisher
FullQueueWanReplication -> WanFullQueuePublisher
WanConfigurationTest -> WanReplicationConfigurationTest
WanBatchReplicationPublisherConfigDTOTest -> WanBatchPublisherConfigDTOTest
CustomCustomWanPublisherConfigDTOTest -> WanCustomPublisherConfigDTOTest
WanBatchReplicationPublisherConfig -> WanBatchPublisherConfig
CustomWanPublisherConfig -> WanCustomPublisherConfig
MapReplicationRemove -> WanMapRemoveEvent
MapReplicationUpdate -> WanMapUpdateEvent
MapReplicationSupportingService -> WanMapSupportingService
WanBatchReplicationPublisherConfigDTO -> WanBatchPublisherConfigDTO
WanReplicationOperation -> WanEventReplicationOperation
ReplicationSupportingService -> WanSupportingService
DelegatingWanReplicationScheme -> DelegatingWanScheme
InternalWanReplicationEvent -> InternalWanEvent
InternalWanReplicationPublisher -> InternalWanPublisher
WanReplicationConsumer -> WanConsumer
WanReplicationEvent -> WanEvent
DistributedServiceWanEventCounters -> WanEventDistributedServiceCounters
MigrationAwareWanReplicationPublisher -> WanMigrationAwarePublisher
WanReplicationPublisher -> WanPublisher
WanReplicationQueueFullException -> WanQueueFullException

Changes in EE:
WanConfigurationContextTest -> WanReplicationConfigurationContextTest
AbstractCacheWanReplicationTest -> AbstractWanPublisherCacheTest
CacheWanBatchReplicationTest -> WanBatchPublisherCacheTest
HdCacheWanBatchReplicationTest -> WanBatchPublisherHDCacheTest
CacheWanReplicationPutAllTest -> WanCachePutAllTest
CacheWanReplicationQuickTest -> WanCacheQuickTest
CacheWanReplicationTestSupport -> WanCacheTestSupport
WanDiscoveryTest -> WanBatchPublisherDiscoveryTest
WanConnectedTest -> WanBatchPublisherConnectedTest
DynamicConfigWANTest -> WanBatchPublisherDynamicConfigTest
HDMapWanSyncTest -> WanBatchPublisherHDMapSyncTest
MapWRHotRestartEnabledTest -> WanBatchPublisherHotRestartMapTest
MapWanReplicationQuickTest -> WanBatchPublisherMapQuickTest
MapWanBatchReplicationSerializationTest -> WanBatchPublisherMapSerializationTest
MapWanBatchReplicationSnapshotTest -> WanBatchPublisherMapSnapshotTest
MapWanBatchReplicationTest -> WanBatchPublisherMapTest
WanReplicationClockDiffTest -> WanClockDiffTest
MapCustomWanReplicationQuickTest -> WanCustomPublisherMapQuickTest
MapWanReplicationTestSupport -> WanMapTestSupport
MultiNodeWanReplicationTest -> WanBatchPublisherMultiNodeTest
WanBatchReplicationPropertiesTest -> WanBatchPublisherPropertiesTest
EWRDataSerializerHookTest -> WanDataSerializerHookTest
WanReplicationTestSupport -> WanTestSupport
CacheReplicationObject -> WanCacheEvent
CacheReplicationRemove -> WanCacheRemoveEvent
CacheReplicationSupportingService -> WanCacheSupportingService
CacheReplicationUpdate -> WanCacheUpdateEvent
EWRBackupAwareOperation -> WanBackupAwareOperation
EWRBaseOperation -> WanBaseOperation
EWRDataSerializerHook -> WanDataSerializerHook
EWRPutBackupOperation -> WanPutBackupOperation
EWRPutOperation -> WanPutOperation
WanBatchReplication -> WanBatchPublisher
WanBatchReplicationProperties -> WanBatchPublisherProperties
BatchWanReplicationEvent -> WanEventBatch
WanReplicationEventQueue -> WanEventQueue
WanReplicationMigrationAwareService -> WanMigrationAwareService
WanReplicationSchemeContainer -> WanSchemeContainer
EnterpriseMapReplicationObject -> WanEnterpriseMapEvent
EnterpriseMapReplicationMerkleTreeNode -> WanEnterpriseMapMerkleTreeNode
EnterpriseMapReplicationRemove -> WanEnterpriseMapRemoveEvent
EnterpriseMapReplicationSupportingService -> WanEnterpriseMapSupportingService
EnterpriseMapReplicationSync -> WanEnterpriseMapSyncEvent
EnterpriseMapReplicationUpdate -> WanEnterpriseMapUpdateEvent
DummyWanReplication -> WanDummyPublisher

Fixes: https://github.com/hazelcast/hazelcast-enterprise/issues/2617
EE: https://github.com/hazelcast/hazelcast-enterprise/pull/3409

mmedenjak commented:
@devozerov true, I had mixed feeling between `WanReplication...` and `Wan...`. Technically speaking, WAN does not necessarily entail "replication" and it _may_ mean something else. But for the sake of brevity, I guess we don't need to force this distinction on every class, attribute and field name.

mmedenjak commented:
@ahmetmircik @blazember @devozerov updated the PR to address more classes and now followed the idea that we avoid the usage of `Replication` since it's implied and just makes the name more verbose. Although, for the sake of making this PR as simple to review and to implement, I've made some restrictions. Please see the updated description.

ahmetmircik commented on hazelcast/src/main/java/com/hazelcast/wan/WanEventDistributedServiceCounters.java:
```diff
@@ -30,7 +30,7 @@
  * This class may contain counters for a single WAN publisher or multiple
  * WAN publishers, depending on its usage.
  */
-public class DistributedServiceWanEventCounters {
```
I think `WanEventCounters` name fit better for public api instead of `DistributedServiceWanEventCounters`. Since it implies a _not distributed_ counterpart but we don't have it. We can rename internal class `com.hazelcast.wan.impl.WanEventCounters` to something else like `WanEventCounterRegistry`.

mmedenjak replied:
Changed both.

mmedenjak commented:
@devozerov @blazember @ahmetmircik addressed all review comments by accepting your suggestions:
- WanReplicationConfig.wanConsumerConfig -> WanReplicationConfig.consumerConfig
- WanBatchPublisherConfig.wanSyncConfig -> WanBatchPublisherConfig.syncConfig
- WanReplicationConfig.addWanBatchReplicationPublisherConfig -> WanReplicationConfig.addBatchReplicationPublisherConfig
- WanEventCounters -> WanEventCounterRegistry
- WanEventDistributedServiceCounters -> WanEventCounters
- WanEventReplicationOperation -> WanEventContainerOperation
- MerkleTreeWanSyncStats -> WanMerkleTreeSyncStats

Renamed getters and setters too. I'll be merging these PRs once they pass tests. If you find some artifact that needs to/could be changed afterwards, let's address it separately.

mmedenjak commented:
Thank you for the reviews, everyone!
